pkg/sync/task/queue/singleWorkerQueue.go

```diff
@@ -0,0 +1,107 @@
+package queue
+
+import (
+	"container/list"
+	"sync"
+
+	"go.uber.org/atomic"
+)
+
+type oneWorkerQueue struct {
+	inUse        atomic.Bool
+	addedToQueue atomic.Bool
+
+	mutex           sync.Mutex
+	queue           *list.List
+	deleteFromQueue func()
+	deleted         bool
+}
+
+func (q *oneWorkerQueue) append(tasks []func()) bool {
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+	if q.deleted {
+		return false
+	}
+	for _, t := range tasks {
+		q.queue.PushBack(t)
+	}
+	return true
+}
+
+func (q *oneWorkerQueue) freeLocked() {
+	q.queue.Init()
+	q.deleteFromQueue()
+	q.deleted = true
+}
+
+func (q *oneWorkerQueue) pop() func() {
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+	if q.queue.Len() > 0 {
+		return q.queue.Remove(q.queue.Front()).(func())
+	}
+	q.freeLocked()
```

Comment:
This could be suboptimal for resources in heavy use, it will keep removing and adding the key to the `q.oneworkerQueue` map. But there's probably no need to optimize it now, we should see it in the profiler if we find a problematic use case.